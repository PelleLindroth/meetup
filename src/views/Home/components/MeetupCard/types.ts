import { Meetup } from '../../../../db/models/Meetup'

export type MeetupCardProps = {
  meetup: Meetup
  upcoming?: boolean
}
