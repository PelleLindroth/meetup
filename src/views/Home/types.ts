import { Meetup } from '../../db/models/Meetup'

export type HomeProps = {
  meetups: Meetup[]
  setMeetups: React.Dispatch<React.SetStateAction<Meetup[]>>
}
