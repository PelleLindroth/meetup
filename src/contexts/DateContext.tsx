import React, { useState } from 'react'

export interface DateContextInterface {
  customDate: Date
  setCustomDate: React.Dispatch<React.SetStateAction<Date>>
}

export const DateContext = React.createContext<DateContextInterface | null>(
  null
)

const DateContextProvider: React.FC = (props) => {
  const [customDate, setCustomDate] = useState<Date>(new Date())

  const dateContext: DateContextInterface = {
    customDate,
    setCustomDate,
  }

  return (
    <DateContext.Provider value={dateContext}>
      {props.children}
    </DateContext.Provider>
  )
}

export default DateContextProvider
