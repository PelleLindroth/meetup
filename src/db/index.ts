import { meetups, Meetup } from './meetups'
import { users, User } from './users'

export const getAllMeetups = (): Meetup[] => {
  const storedMeetups: Meetup[] = JSON.parse(localStorage.getItem('meetups')!)
  let returnedMeetups: Meetup[] = []

  if (storedMeetups) {
    returnedMeetups.push(...storedMeetups)
  } else returnedMeetups.push(...meetups)

  returnedMeetups.sort((a, b) => {
    if (a.date.getTime() > b.date.getTime()) {
      return 1
    } else if (b.date.getTime() > a.date.getTime()) {
      return -1
    } else {
      return 0
    }
  })

  return returnedMeetups
}

export const getMockMeetups = () => {
  meetups.sort((a, b) => {
    if (a.date.getTime() > b.date.getTime()) {
      return 1
    } else if (b.date.getTime() > a.date.getTime()) {
      return -1
    } else {
      return 0
    }
  })

  return meetups
}

export const saveMeetupToLocalStorage = (meetup: Meetup) => {
  const storedMeetups: Meetup[] = JSON.parse(localStorage.getItem('meetups')!)

  if (storedMeetups) {
    storedMeetups.push(meetup)

    localStorage.setItem('meetups', JSON.stringify(storedMeetups))
  } else {
    localStorage.setItem('meetups', JSON.stringify([meetup]))
  }
}

export const getMeetupById = (id: string): Meetup | undefined => {
  return meetups.find((meetup) => meetup.id === id)
}

export const validateUser = (email: string, password: string): User | null => {
  const user = users.find((user) => user.email === email)

  if (user) {
    return user.password === password ? user : null
  } else {
    return null
  }
}

export const getUserById = (id: string): User | undefined => {
  return users.find((user) => user.id === id)
}

export const getMockUsers = () => {
  return users
}
