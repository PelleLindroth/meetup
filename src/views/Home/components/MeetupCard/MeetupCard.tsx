import { Link } from 'react-router-dom'
import Cloud from '../../../../assets/icons/cloud.png'
import Time from '../../../../assets/icons/time.png'
import Pin from '../../../../assets/icons/pin.png'
import Arrow from '../../../../assets/icons/arrow.png'
import styles from './MeetupCard.module.scss'
import { MeetupCardProps } from './types'

const MeetupCard = (props: MeetupCardProps) => {
  const { meetup, upcoming } = props

  return (
    <li className={styles.cardWrapper}>
      <div
        title={upcoming ? 'upcoming-event-header' : 'past-event-header'}
        style={{ backgroundColor: upcoming ? '#0E81C9' : '#DCDDDE' }}
        className={styles.cardHeader}
      >
        <h2>{meetup.title}</h2>
      </div>
      <div className={styles.cardBody}>
        <div className={styles.leftColumn}>
          <div className={styles.dateRow}>
            <img src={Time} alt="Time icon" className={styles.icon} />
            <h3 title="date">
              {new Intl.DateTimeFormat('en-GB', {
                year: 'numeric',
                month: 'long',
                day: '2-digit',
                hour: 'numeric',
                minute: 'numeric',
              }).format(meetup.date)}{' '}
              CET
            </h3>
          </div>
          <div className={styles.locationRow}>
            {meetup.online ? (
              <>
                <img src={Cloud} alt="Cloud icon" className={styles.icon} />
                <h3 title="location">Online event</h3>
              </>
            ) : (
              <>
                <img src={Pin} alt="Map pin icon" className={styles.icon} />
                <h3 title="location">{`${meetup.location?.street}, ${meetup.location?.city}`}</h3>
              </>
            )}
          </div>
          <p title="capacity" className={styles.capacityRow}>
            <strong>Capacity: </strong>
            {meetup.capacity ? (
              <>
                <span>{`${meetup.capacity}, `}</span>
                <span
                  className={
                    meetup.capacity - meetup.attending > 0
                      ? styles.available
                      : styles.unavailable
                  }
                >{`${meetup.capacity - meetup.attending} available`}</span>
              </>
            ) : (
              <span className={styles.available}>Unlimited</span>
            )}
          </p>
        </div>

        <div className={styles.rightColumn}>
          <Link to={`/meetup/${meetup.id}`}>
            <img
              src={Arrow}
              alt="Arrow icon"
              className={styles.arrowIcon}
              title="See meetup details"
            />
          </Link>
        </div>
      </div>
    </li>
  )
}

export default MeetupCard
