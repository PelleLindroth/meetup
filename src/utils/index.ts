import { User } from '../db/users'
import { Meetup } from '../db/meetups'

export const formatDate = (date: Date) => {
  return `${new Intl.DateTimeFormat('en-GB', {
    year: 'numeric',
    month: 'long',
    day: '2-digit',
    hour: 'numeric',
    minute: 'numeric',
  }).format(date)}`
}

const sortChronologically = (array: Meetup[]) => {
  return array.sort((a, b) => a.date.getTime() - b.date.getTime())
}

export const createMeetupList = (
  meetups: Meetup[],
  listType: string,
  searchPhrase: string,
  searchFilter: string
) => {
  const meetupList = meetups.filter((meetup) => {
    if (listType === 'past') {
      return (
        meetup.date.getTime() < Date.now() &&
        meetup.title.toLowerCase().includes(searchPhrase.toLowerCase()) &&
        (searchFilter === listType || searchFilter === 'all')
      )
    } else {
      return (
        meetup.date.getTime() > Date.now() &&
        meetup.title.toLowerCase().includes(searchPhrase.toLowerCase()) &&
        (searchFilter === listType || searchFilter === 'all')
      )
    }
  })

  return sortChronologically(meetupList)
}

export const createUserLists = (user: User, meetups: Meetup[]) => {
  return {
    ownUpcoming: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          meetup.arranger.id === user!.id &&
          meetup.date.getTime() > new Date().getTime()
      )
    ),
    ownPast: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          meetup.arranger.id === user!.id &&
          meetup.date.getTime() < new Date().getTime()
      )
    ),
    upcomingAttending: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          user!.attending.includes(meetup.id) &&
          meetup.date.getTime() > new Date().getTime()
      )
    ),
    pastAttended: sortChronologically(
      meetups.filter(
        (meetup: Meetup) =>
          user!.attending.includes(meetup.id) &&
          meetup.date.getTime() < new Date().getTime()
      )
    ),
  }
}
