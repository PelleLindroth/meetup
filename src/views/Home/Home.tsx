import { Meetup } from '../../db/meetups'
import MeetupList from '../../components/MeetupList'

type HomeProps = {
  meetups: Meetup[]
  setMeetups: React.Dispatch<React.SetStateAction<Meetup[]>>
}

const Home = (props: HomeProps) => {
  const { meetups } = props

  return (
    <section>
      <h2>UPCOMING EVENTS</h2>
      <MeetupList
        meetups={meetups.filter((meetup) => meetup.date.getTime() > Date.now())}
        upcoming
      />
      <h2>PAST EVENTS</h2>
      <MeetupList
        meetups={meetups.filter((meetup) => meetup.date.getTime() < Date.now())}
      />
    </section>
  )
}

export default Home
