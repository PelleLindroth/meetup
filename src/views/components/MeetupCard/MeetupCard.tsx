import { Link } from 'react-router-dom'
import { useLocation } from 'react-router'
import { formatDate } from '../../../utils'
import CloudIcon from '../../../assets/icons/cloud.png'
import TimeIcon from '../../../assets/icons/time.png'
import PinIcon from '../../..//assets/icons/pin.png'
import ArrowIcon from '../../../assets/icons/arrow.png'
import styles from './MeetupCard.module.scss'
import { MeetupCardProps } from './types'

const MeetupCard = (props: MeetupCardProps) => {
  const location = useLocation()
  const { meetup, upcoming } = props

  return (
    <li aria-label="meetup-card" className={styles.cardWrapper}>
      <div
        title={upcoming ? 'upcoming-event-header' : 'past-event-header'}
        style={{ backgroundColor: upcoming ? '#0E81C9' : '#DCDDDE' }}
        className={styles.cardHeader}
      >
        <h2>{meetup.title}</h2>
      </div>
      <div className={styles.cardBody}>
        <div className={styles.leftColumn}>
          <div className={styles.dateRow}>
            <img src={TimeIcon} alt="Time icon" className={styles.icon} />
            <h3 title="date">{formatDate(meetup.date)} CET</h3>
          </div>
          <div className={styles.locationRow}>
            {meetup.online ? (
              <>
                <img src={CloudIcon} alt="Cloud icon" className={styles.icon} />
                <h3 title="location">
                  {location.pathname.match(/^\/profile\/[0-9]$/) ? (
                    <a target="_blank" rel="noreferrer" href={meetup.url!}>
                      {meetup.url}
                    </a>
                  ) : (
                    'Online event'
                  )}
                </h3>
              </>
            ) : (
              <>
                <img src={PinIcon} alt="Map pin icon" className={styles.icon} />
                <h3 title="location">{`${meetup.location?.street}, ${meetup.location?.city}`}</h3>
              </>
            )}
          </div>
          <p title="capacity" className={styles.capacityRow}>
            <strong>Capacity: </strong>
            {meetup.capacity ? (
              <>
                <span>{`${meetup.capacity}, `}</span>
                <span
                  className={
                    meetup.capacity - meetup.attending > 0
                      ? styles.available
                      : styles.unavailable
                  }
                >{`${meetup.capacity - meetup.attending} available`}</span>
              </>
            ) : (
              <span className={styles.available}>Unlimited</span>
            )}
          </p>
        </div>

        <div className={styles.rightColumn}>
          <Link to={`/meetup/${meetup.id}`}>
            <img
              src={ArrowIcon}
              alt="Arrow icon"
              className={styles.arrowIcon}
              title="See meetup details"
            />
          </Link>
        </div>
      </div>
    </li>
  )
}

export default MeetupCard
