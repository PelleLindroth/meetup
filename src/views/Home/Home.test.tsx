import { mount } from 'enzyme'
import { screen } from '@testing-library/react'
import { Link, MemoryRouter } from 'react-router-dom'
import { DateContext, DateContextInterface } from '../../contexts/DateContext'
import { renderWithRouter } from '../../utils/testing-utils'
import Home from './Home'
import MeetupCard from './components/MeetupCard'
import { getMockMeetups } from '../../db'
import userEvent from '@testing-library/user-event'

const mockDateContext: DateContextInterface = {
  customDate: new Date(1640354400000),
  setCustomDate: jest.fn(),
}

describe('Home', () => {
  it('mounts Home view correctly with meetups', () => {
    const wrapper = mount(
      <DateContext.Provider value={mockDateContext}>
        <MemoryRouter>
          <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
        </MemoryRouter>
      </DateContext.Provider>
    )
    expect(wrapper.find(Home)).toMatchSnapshot()
  })
  it('renders a list of meetups', async () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const meetupList = screen.getAllByRole('listitem')

    expect(meetupList).toHaveLength(5)
  })
  it('renders upcoming meetups chronologically (ascending)', async () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const dates = screen.getAllByTitle('date')

    expect(dates[0]).toHaveTextContent('10 January 2022')
    expect(dates[2]).toHaveTextContent('02 July 2022')
  })
  it('renders an Upcoming Events list and a Past Events list according to current date', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const upcomingEventsList = screen.getByTestId('upcoming-events')
    const pastEventsList = screen.getByTestId('past-events')

    expect(upcomingEventsList).toBeInTheDocument()
    expect(pastEventsList).toBeInTheDocument()
    expect(upcomingEventsList.childNodes).toHaveLength(3)
    expect(pastEventsList.childNodes).toHaveLength(2)
  })
  it('separates upcoming and past events by background color on card headers', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const upcomingEventHeaders = screen.getAllByTitle('upcoming-event-header')
    const pastEventHeaders = screen.getAllByTitle('past-event-header')

    expect(upcomingEventHeaders[0]).toHaveStyle({ backgroundColor: '#0E81C9' })
    expect(pastEventHeaders[0]).toHaveStyle({ backgroundColor: '#DCDDDE' })
  })
  it('renders separate meetup card for every meetup with title, date, location, capacity and an img link to detail page', () => {
    const wrapper = mount(
      <DateContext.Provider value={mockDateContext}>
        <MemoryRouter>
          <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
        </MemoryRouter>
      </DateContext.Provider>
    )
    const meetupCard = wrapper.find(MeetupCard).at(0)

    expect(meetupCard.find('h2').text()).toBe('Rust course')
    expect(meetupCard.find('h3').at(0).text()).toBe(
      '10 January 2022, 11:00 CET'
    )
    expect(meetupCard.find('h3').at(1).text()).toMatch(/online event/i)
    expect(meetupCard.find('p').text()).toMatch('Capacity: Unlimited')
    expect(meetupCard.find(Link).find('.arrowIcon').type()).toBe('img')
    expect(meetupCard).toMatchSnapshot()
  })
  it('renders a search input field, empty by default', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const searchBox = screen.getByRole('searchbox')

    expect(searchBox).toBeInTheDocument()
    expect(searchBox).toHaveTextContent('')
  })
  it('renders a filter select input with "all" as default', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const selectInput = screen.getByRole('combobox')

    expect(selectInput).toBeInTheDocument()
    expect(selectInput).toHaveValue('all')
  })
  it('shows filtered events when typing in search field', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const searchBox = screen.getByRole('searchbox')

    userEvent.type(searchBox, 'pizza')

    const meetupList = screen.getAllByRole('listitem')

    expect(meetupList).toHaveLength(1)
    expect(meetupList[0]).toHaveTextContent(/pizza/i)
  })
  it('shows only past events when pastEvents filter is active', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const filterSelect = screen.getByRole('combobox')

    userEvent.selectOptions(filterSelect, 'past')

    const meetupList = screen.getAllByRole('listitem')

    expect(meetupList).toHaveLength(2)
    expect(meetupList[0]).toHaveTextContent(/pizza/i)
  })
  it('shows only upcoming events when upcomingEvents filter is active', () => {
    renderWithRouter(
      <DateContext.Provider value={mockDateContext}>
        <Home meetups={getMockMeetups()} setMeetups={jest.fn()} />
      </DateContext.Provider>
    )

    const filterSelect = screen.getByRole('combobox')

    userEvent.selectOptions(filterSelect, 'upcoming')

    const meetupList = screen.getAllByRole('listitem')

    expect(meetupList).toHaveLength(3)
    expect(meetupList[0]).toHaveTextContent(/rust/i)
  })
})
