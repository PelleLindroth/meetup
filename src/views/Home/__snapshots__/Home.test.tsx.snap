// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home mounts Home view correctly with meetups 1`] = `
<Home
  meetups={
    Array [
      Object {
        "arranger": Object {
          "attended": Array [],
          "attending": Array [],
          "email": "tjalle@yahoo.com",
          "firstName": "Tjalle",
          "id": "1",
          "lastName": "Bergkvist",
          "password": "grillkorv",
        },
        "capacity": null,
        "comments": Array [],
        "date": 2021-08-28T17:00:00.000Z,
        "description": "We meet in the park with pizzas and hang out all night",
        "id": "1",
        "location": Object {
          "city": "Stockholm",
          "country": "Sweden",
          "street": "Vitabergsparken",
        },
        "online": false,
        "reviews": Array [],
        "title": "Pizza Picnic",
        "url": "https://www.udemy.com/course/rust-lang/",
      },
      Object {
        "arranger": Object {
          "attended": Array [],
          "attending": Array [],
          "email": "babsan@yahoo.com",
          "firstName": "Barbro",
          "id": "3",
          "lastName": "Lundborg",
          "password": "password123",
        },
        "capacity": null,
        "comments": Array [],
        "date": 2022-01-10T10:00:00.000Z,
        "description": "Learn a new programming language from the ground up",
        "id": "2",
        "online": true,
        "reviews": Array [],
        "title": "Rust course",
        "url": "https://www.udemy.com/course/rust-lang/",
      },
      Object {
        "arranger": Object {
          "attended": Array [],
          "attending": Array [],
          "email": "kenta@yahoo.com",
          "firstName": "Kenta",
          "id": "2",
          "lastName": "Andersson",
          "password": "bananpaj",
        },
        "availableSeats": 39567,
        "capacity": 43883,
        "comments": Array [],
        "date": 2022-05-13T16:00:00.000Z,
        "description": "A classic football match between the south and the north",
        "id": "3",
        "location": Object {
          "city": "41005 Sevilla",
          "country": "Spain",
          "street": "C. Sevilla Fútbol Club",
        },
        "online": false,
        "reviews": Array [],
        "title": "Sevilla - Real Sociedad",
      },
      Object {
        "arranger": Object {
          "attended": Array [],
          "attending": Array [],
          "email": "loffe@yahoo.com",
          "firstName": "Janne",
          "id": "4",
          "lastName": "Karlsson",
          "password": "storstark",
        },
        "capacity": null,
        "comments": Array [],
        "date": 2022-07-02T15:00:00.000Z,
        "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
        "id": "4",
        "location": Object {
          "city": "Stockholm",
          "country": "Sweden",
          "street": "Gärdet",
        },
        "online": false,
        "reviews": Array [],
        "title": "Kite festival",
      },
    ]
  }
  setMeetups={[MockFunction]}
>
  <section>
    <h2>
      UPCOMING EVENTS
    </h2>
    <MeetupList
      meetups={
        Array [
          Object {
            "arranger": Object {
              "attended": Array [],
              "attending": Array [],
              "email": "babsan@yahoo.com",
              "firstName": "Barbro",
              "id": "3",
              "lastName": "Lundborg",
              "password": "password123",
            },
            "capacity": null,
            "comments": Array [],
            "date": 2022-01-10T10:00:00.000Z,
            "description": "Learn a new programming language from the ground up",
            "id": "2",
            "online": true,
            "reviews": Array [],
            "title": "Rust course",
            "url": "https://www.udemy.com/course/rust-lang/",
          },
          Object {
            "arranger": Object {
              "attended": Array [],
              "attending": Array [],
              "email": "kenta@yahoo.com",
              "firstName": "Kenta",
              "id": "2",
              "lastName": "Andersson",
              "password": "bananpaj",
            },
            "availableSeats": 39567,
            "capacity": 43883,
            "comments": Array [],
            "date": 2022-05-13T16:00:00.000Z,
            "description": "A classic football match between the south and the north",
            "id": "3",
            "location": Object {
              "city": "41005 Sevilla",
              "country": "Spain",
              "street": "C. Sevilla Fútbol Club",
            },
            "online": false,
            "reviews": Array [],
            "title": "Sevilla - Real Sociedad",
          },
          Object {
            "arranger": Object {
              "attended": Array [],
              "attending": Array [],
              "email": "loffe@yahoo.com",
              "firstName": "Janne",
              "id": "4",
              "lastName": "Karlsson",
              "password": "storstark",
            },
            "capacity": null,
            "comments": Array [],
            "date": 2022-07-02T15:00:00.000Z,
            "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
            "id": "4",
            "location": Object {
              "city": "Stockholm",
              "country": "Sweden",
              "street": "Gärdet",
            },
            "online": false,
            "reviews": Array [],
            "title": "Kite festival",
          },
        ]
      }
      upcoming={true}
    >
      <ul
        className="meetupList"
        data-testid="upcoming-events"
      >
        <MeetupCard
          key="2"
          meetup={
            Object {
              "arranger": Object {
                "attended": Array [],
                "attending": Array [],
                "email": "babsan@yahoo.com",
                "firstName": "Barbro",
                "id": "3",
                "lastName": "Lundborg",
                "password": "password123",
              },
              "capacity": null,
              "comments": Array [],
              "date": 2022-01-10T10:00:00.000Z,
              "description": "Learn a new programming language from the ground up",
              "id": "2",
              "online": true,
              "reviews": Array [],
              "title": "Rust course",
              "url": "https://www.udemy.com/course/rust-lang/",
            }
          }
          upcoming={true}
        >
          <li
            className="cardWrapper"
          >
            <div
              className="cardHeader"
              style={
                Object {
                  "backgroundColor": "#0E81C9",
                }
              }
              title="upcoming-event-header"
            >
              <h2>
                Rust course
              </h2>
            </div>
            <div
              className="cardBody"
            >
              <div
                className="leftColumn"
              >
                <div
                  className="dateRow"
                >
                  <img
                    alt="Time icon"
                    className="icon"
                    src="time.png"
                  />
                  <h3
                    title="date"
                  >
                    10 January 2022, 11:00
                     
                    CET
                  </h3>
                </div>
                <div
                  className="locationRow"
                >
                  <img
                    alt="Cloud icon"
                    className="icon"
                    src="cloud.png"
                  />
                  <h3
                    title="location"
                  >
                    Online event
                  </h3>
                </div>
                <p
                  title="capacity"
                >
                  Capacity:
                   
                  <em>
                    unlimited
                  </em>
                </p>
              </div>
              <div
                className="rightColumn"
              >
                <Link
                  to="/meetup/2"
                >
                  <a
                    href="/meetup/2"
                    onClick={[Function]}
                  >
                    <img
                      alt="Arrow icon"
                      className="arrowIcon"
                      src="arrow.png"
                      title="See meetup details"
                    />
                  </a>
                </Link>
              </div>
            </div>
          </li>
        </MeetupCard>
        <MeetupCard
          key="3"
          meetup={
            Object {
              "arranger": Object {
                "attended": Array [],
                "attending": Array [],
                "email": "kenta@yahoo.com",
                "firstName": "Kenta",
                "id": "2",
                "lastName": "Andersson",
                "password": "bananpaj",
              },
              "availableSeats": 39567,
              "capacity": 43883,
              "comments": Array [],
              "date": 2022-05-13T16:00:00.000Z,
              "description": "A classic football match between the south and the north",
              "id": "3",
              "location": Object {
                "city": "41005 Sevilla",
                "country": "Spain",
                "street": "C. Sevilla Fútbol Club",
              },
              "online": false,
              "reviews": Array [],
              "title": "Sevilla - Real Sociedad",
            }
          }
          upcoming={true}
        >
          <li
            className="cardWrapper"
          >
            <div
              className="cardHeader"
              style={
                Object {
                  "backgroundColor": "#0E81C9",
                }
              }
              title="upcoming-event-header"
            >
              <h2>
                Sevilla - Real Sociedad
              </h2>
            </div>
            <div
              className="cardBody"
            >
              <div
                className="leftColumn"
              >
                <div
                  className="dateRow"
                >
                  <img
                    alt="Time icon"
                    className="icon"
                    src="time.png"
                  />
                  <h3
                    title="date"
                  >
                    13 May 2022, 18:00
                     
                    CET
                  </h3>
                </div>
                <div
                  className="locationRow"
                >
                  <img
                    alt="Pin icon"
                    className="icon"
                    src="pin.png"
                  />
                  <h3
                    title="location"
                  >
                    C. Sevilla Fútbol Club, 41005 Sevilla
                  </h3>
                </div>
                <p
                  title="capacity"
                >
                  Capacity:
                   
                  <em>
                    43883, 39567 available
                  </em>
                </p>
              </div>
              <div
                className="rightColumn"
              >
                <Link
                  to="/meetup/3"
                >
                  <a
                    href="/meetup/3"
                    onClick={[Function]}
                  >
                    <img
                      alt="Arrow icon"
                      className="arrowIcon"
                      src="arrow.png"
                      title="See meetup details"
                    />
                  </a>
                </Link>
              </div>
            </div>
          </li>
        </MeetupCard>
        <MeetupCard
          key="4"
          meetup={
            Object {
              "arranger": Object {
                "attended": Array [],
                "attending": Array [],
                "email": "loffe@yahoo.com",
                "firstName": "Janne",
                "id": "4",
                "lastName": "Karlsson",
                "password": "storstark",
              },
              "capacity": null,
              "comments": Array [],
              "date": 2022-07-02T15:00:00.000Z,
              "description": "Classic Kite festival at Gärdet, bring your own kite or just watch the pros",
              "id": "4",
              "location": Object {
                "city": "Stockholm",
                "country": "Sweden",
                "street": "Gärdet",
              },
              "online": false,
              "reviews": Array [],
              "title": "Kite festival",
            }
          }
          upcoming={true}
        >
          <li
            className="cardWrapper"
          >
            <div
              className="cardHeader"
              style={
                Object {
                  "backgroundColor": "#0E81C9",
                }
              }
              title="upcoming-event-header"
            >
              <h2>
                Kite festival
              </h2>
            </div>
            <div
              className="cardBody"
            >
              <div
                className="leftColumn"
              >
                <div
                  className="dateRow"
                >
                  <img
                    alt="Time icon"
                    className="icon"
                    src="time.png"
                  />
                  <h3
                    title="date"
                  >
                    02 July 2022, 17:00
                     
                    CET
                  </h3>
                </div>
                <div
                  className="locationRow"
                >
                  <img
                    alt="Pin icon"
                    className="icon"
                    src="pin.png"
                  />
                  <h3
                    title="location"
                  >
                    Gärdet, Stockholm
                  </h3>
                </div>
                <p
                  title="capacity"
                >
                  Capacity:
                   
                  <em>
                    unlimited
                  </em>
                </p>
              </div>
              <div
                className="rightColumn"
              >
                <Link
                  to="/meetup/4"
                >
                  <a
                    href="/meetup/4"
                    onClick={[Function]}
                  >
                    <img
                      alt="Arrow icon"
                      className="arrowIcon"
                      src="arrow.png"
                      title="See meetup details"
                    />
                  </a>
                </Link>
              </div>
            </div>
          </li>
        </MeetupCard>
      </ul>
    </MeetupList>
    <h2>
      PAST EVENTS
    </h2>
    <MeetupList
      meetups={
        Array [
          Object {
            "arranger": Object {
              "attended": Array [],
              "attending": Array [],
              "email": "tjalle@yahoo.com",
              "firstName": "Tjalle",
              "id": "1",
              "lastName": "Bergkvist",
              "password": "grillkorv",
            },
            "capacity": null,
            "comments": Array [],
            "date": 2021-08-28T17:00:00.000Z,
            "description": "We meet in the park with pizzas and hang out all night",
            "id": "1",
            "location": Object {
              "city": "Stockholm",
              "country": "Sweden",
              "street": "Vitabergsparken",
            },
            "online": false,
            "reviews": Array [],
            "title": "Pizza Picnic",
            "url": "https://www.udemy.com/course/rust-lang/",
          },
        ]
      }
    >
      <ul
        className="meetupList"
        data-testid="past-events"
      >
        <MeetupCard
          key="1"
          meetup={
            Object {
              "arranger": Object {
                "attended": Array [],
                "attending": Array [],
                "email": "tjalle@yahoo.com",
                "firstName": "Tjalle",
                "id": "1",
                "lastName": "Bergkvist",
                "password": "grillkorv",
              },
              "capacity": null,
              "comments": Array [],
              "date": 2021-08-28T17:00:00.000Z,
              "description": "We meet in the park with pizzas and hang out all night",
              "id": "1",
              "location": Object {
                "city": "Stockholm",
                "country": "Sweden",
                "street": "Vitabergsparken",
              },
              "online": false,
              "reviews": Array [],
              "title": "Pizza Picnic",
              "url": "https://www.udemy.com/course/rust-lang/",
            }
          }
          upcoming={false}
        >
          <li
            className="cardWrapper"
          >
            <div
              className="cardHeader"
              style={
                Object {
                  "backgroundColor": "#DCDDDE",
                }
              }
              title="past-event-header"
            >
              <h2>
                Pizza Picnic
              </h2>
            </div>
            <div
              className="cardBody"
            >
              <div
                className="leftColumn"
              >
                <div
                  className="dateRow"
                >
                  <img
                    alt="Time icon"
                    className="icon"
                    src="time.png"
                  />
                  <h3
                    title="date"
                  >
                    28 August 2021, 19:00
                     
                    CET
                  </h3>
                </div>
                <div
                  className="locationRow"
                >
                  <img
                    alt="Pin icon"
                    className="icon"
                    src="pin.png"
                  />
                  <h3
                    title="location"
                  >
                    Vitabergsparken, Stockholm
                  </h3>
                </div>
                <p
                  title="capacity"
                >
                  Capacity:
                   
                  <em>
                    unlimited
                  </em>
                </p>
              </div>
              <div
                className="rightColumn"
              >
                <Link
                  to="/meetup/1"
                >
                  <a
                    href="/meetup/1"
                    onClick={[Function]}
                  >
                    <img
                      alt="Arrow icon"
                      className="arrowIcon"
                      src="arrow.png"
                      title="See meetup details"
                    />
                  </a>
                </Link>
              </div>
            </div>
          </li>
        </MeetupCard>
      </ul>
    </MeetupList>
  </section>
</Home>
`;

exports[`Home renders separate meetup card for every meetup with title, date, location, capacity and an img link to detail page 1`] = `
<MeetupCard
  key="2"
  meetup={
    Object {
      "arranger": Object {
        "attended": Array [],
        "attending": Array [],
        "email": "babsan@yahoo.com",
        "firstName": "Barbro",
        "id": "3",
        "lastName": "Lundborg",
        "password": "password123",
      },
      "capacity": null,
      "comments": Array [],
      "date": 2022-01-10T10:00:00.000Z,
      "description": "Learn a new programming language from the ground up",
      "id": "2",
      "online": true,
      "reviews": Array [],
      "title": "Rust course",
      "url": "https://www.udemy.com/course/rust-lang/",
    }
  }
  upcoming={true}
>
  <li
    className="cardWrapper"
  >
    <div
      className="cardHeader"
      style={
        Object {
          "backgroundColor": "#0E81C9",
        }
      }
      title="upcoming-event-header"
    >
      <h2>
        Rust course
      </h2>
    </div>
    <div
      className="cardBody"
    >
      <div
        className="leftColumn"
      >
        <div
          className="dateRow"
        >
          <img
            alt="Time icon"
            className="icon"
            src="time.png"
          />
          <h3
            title="date"
          >
            10 January 2022, 11:00
             
            CET
          </h3>
        </div>
        <div
          className="locationRow"
        >
          <img
            alt="Cloud icon"
            className="icon"
            src="cloud.png"
          />
          <h3
            title="location"
          >
            Online event
          </h3>
        </div>
        <p
          title="capacity"
        >
          Capacity:
           
          <em>
            unlimited
          </em>
        </p>
      </div>
      <div
        className="rightColumn"
      >
        <Link
          to="/meetup/2"
        >
          <a
            href="/meetup/2"
            onClick={[Function]}
          >
            <img
              alt="Arrow icon"
              className="arrowIcon"
              src="arrow.png"
              title="See meetup details"
            />
          </a>
        </Link>
      </div>
    </div>
  </li>
</MeetupCard>
`;
